app/src/main/java/com/prac/githubrepo/login/LoginViewModel.kt

```diff
@@ -1,7 +1,44 @@
 package com.prac.githubrepo.login
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.prac.data.exception.GitHubApiException
+import com.prac.data.repository.TokenRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
 
-class LoginViewModel : ViewModel() {
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val tokenRepository: TokenRepository
+): ViewModel() {
+    private val _uiState = MutableStateFlow(LoginUIState())
+    val uiState = _uiState.asStateFlow()
+
+    private val _accessToken = MutableStateFlow<String?>(null)
+    val accessToken = _accessToken.asStateFlow()
+
+    fun fetchAccessToken(clientID: String, clientSecret: String, code: String) {
+        viewModelScope.launch {
+            _uiState.value = LoginUIState(isLoading = true)
```

Comment:
LoginUIState 를 매번 생성해서 발행하기보다, _uiState.update 를 하는 편을 권장드려볼게요!
```suggestion
            _uiState.update {
                it.copy(isLoading = true)
            }
```